react/AppIcon/test/AppIcon.spec.js

```diff
@@ -114,49 +113,4 @@ describe('AppIcon component', () => {
       />
     )
   })
-
-  it(`uses Preloader.preload when no fetchIcon method is provided`, async done => {
-    jest.mock('../Preloader')
-    const AppIcon = require('../').AppIcon
-    const Preloader = require('../Preloader')
-    const wrapper = shallow(
-      <AppIcon
-        app={app}
-        client={mockClient}
-        onReady={() => {
-          wrapper.update()
-          const component = wrapper.getElement()
-          expect(component).toMatchSnapshot()
-          expect(Preloader.preload).toHaveBeenCalledWith(app, domain, secure)
-          expect(console.error).not.toHaveBeenCalled()
-          done()
-        }}
-      />
-    )
-  })
-
-  it(`renders immediately when icon is alredy preloaded`, async done => {
-    jest.mock('../Preloader')
-    const AppIcon = require('../').AppIcon
-    const Preloader = require('../Preloader')
-
-    shallow(
-      <AppIcon
-        app={app}
-        client={mockClient}
-        onReady={() => {
-          const wrapper = shallow(<AppIcon app={app} client={mockClient} />)
-          const component = wrapper.getElement()
-          expect(component).toMatchSnapshot()
-          expect(Preloader.getPreloaded).toHaveBeenCalledWith(
-            app,
-            domain,
-            secure
-          )
-          expect(console.error).not.toHaveBeenCalled()
-          done()
-        }}
-      />
-    )
-  })
```

Comment:
Now we don't have a preloaded state, if the icon is already in cache, we will have a `FETCHING` status then the `done` one. 

So we'll have a kind of visual glitch with a quick switch between the loader and the image. 

I'm wondering if we should consider that as a regression or not.

Reply:
Ce n'est pas visible dans mes tests, il n'y a pas de glitch avec un switch rapide entre le loader et l'image quand il y a l'image en cache. Je suppose que React traite tout dans la foulée et malgré les différents rerender n'affiche que le resultat final, étant donnée qu'il n'y a pas d'attente due à de l'asynchrone ?

Petite correction supplémentaire, on avait un différence de comportement à l'init du composant. S'il n'y a pas de `client`, il doit s'init avec un status `ERRORED` sinon avec un status `FETCHING`. C'est maintenant corrigé.